x/sponsorship/types/msgs.go

```diff
@@ -0,0 +1,50 @@
+package types
+
+import (
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	sdkerrors "github.com/cosmos/cosmos-sdk/types/errors"
+)
+
+func (m MsgVote) ValidateBasic() error {
+	_, err := sdk.AccAddressFromBech32(m.Voter)
+	if err != nil {
+		return sdkerrors.ErrInvalidAddress.Wrapf(
+			"voter '%s' must be a valid bech32 address: %s",
+			m.Voter, err.Error(),
+		)
+	}
+
+	err = ValidateGaugeWeights(m.Weights)
+	if err != nil {
+		return ErrInvalidDistribution.Wrap(err.Error())
+	}
+
+	return nil
+}
+
+func (m MsgVote) GetSigners() []sdk.AccAddress {
+	signer, _ := sdk.AccAddressFromBech32(m.Voter)
+	return []sdk.AccAddress{signer}
+}
+
+func (m MsgUpdateParams) ValidateBasic() error {
```

Comment:
why we need dedicated `MsgUpdateParams`? isn't params can be upgrade using the generic `update-params` proposal?